### PR TITLE
Plugin: Fix building plugin zip to include interactive blocks

### PR DESCRIPTION
**Pull request by gziolo**
## What?

Fixes the issue reported in the https://github.com/WordPress/gutenberg/pull/50041#issuecomment-1544003629:

> Version 15.9.20230511 returns 404 for the following files:
> ```
> vendors.min.js
> interactivity.min.js
> navigation.min.js
> ```

## Why?

Files were missing in the `gutenberg.zip` file.

## Testing Instructions

Run `npm run build:plugin-zip locally` (_warning: it removes all artifacts from the branch_) or check one of the CI job in this branch that should have the zip file attached.

I uploaded the `gutenberg.zip` file to a WordPress instance and ensured that the Navigation block works correctly with the feature flag enabled:

Basically, we have to ensure that the Navigation block, the navigation submenu, and the page list behave exactly the same when the option is activated/deactivated. I used the testing instructions from #50041:

> Toggle the Experimental option in Gutenberg -> Experiments:
> - Check that the mentioned experiences (and anything not mentioned) work as expected.
> - Check that it is only sending the Interactivity API scripts.
> - Change the settings of the navigation block to ensure it works fine in all cases.
> - If the Overlay menu is set to off and the "Show arrow" is set to false, the scripts shouldn't be loaded.

### CODE REVIEW

**Comment by gziolo** [comment]
@fabiankaegy, we must backport this bug fix to the Gutenberg 15.8 RC.

**Review by ntwb — APPROVED** [review]

**Comment by fabiankaegy** [comment]
I just cherry-picked this PR to the release/15.8 branch to get it included in the next release: 4871f27f48
